dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java

```diff
@@ -312,16 +316,20 @@ private void waitForPermanentStorageTransfer(final long totalJobSize) throws IOE
                 resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString(), LogType.SUCCESS);
 
         if (isCacheJobEnable) {
-            while (!ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult().getStatus().toString().equals(StringConstants.JOB_COMPLETED)) {
-                Thread.sleep(60000);
-            }
-
-            LOG.info("Job transferred to permanent storage location");
-            final String newDate = DateFormat.formatDate(new Date());
-
-            loggingService.logMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(),
-                    FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, newDate,
-                    resourceBundle.getString("permanentStorageLocation"), false).toString(), LogType.SUCCESS);
+            Observable.interval(60, TimeUnit.SECONDS)
+                    .observeOn(JavaFxScheduler.platform())
+                    .takeUntil( event -> ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult().getStatus() == JobStatus.COMPLETED)
+                    .doOnComplete(() -> {
+                        LOG.info("Job transferred to permanent storage location");
+
+                        final String newDate = DateFormat.formatDate(new Date());
+                        loggingService.logMessage(
+                            StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(),
+                            FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, newDate,
+                            resourceBundle.getString("permanentStorageLocation"), false).toString(), LogType.SUCCESS);
+
+                        Platform.runLater(() -> Ds3PanelService.refresh(remoteDestination));
```

Comment:
As far as I can tell, this should already be on the JavaFx GUI thread.  I'm not sure the `Platform.runLater()` is required here.